central/convert/internaltov2storage/utils.go

```diff
@@ -0,0 +1,31 @@
+package internaltov2storage
+
+import (
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// BuildProfileRefID returns a UUID based off cluster, profileID, and product type
+func BuildProfileRefID(clusterID string, profileID string, productType string) string {
```

Comment:
I can see how this simplifies doing joins but carries the same disadvantages as a cache.
That means, when a some of the data in `BuildProfileRefID` needs to change this link returns another result, that would mean in case of a UPDATE operations this is not deterministically anymore.

I've tried to enhance the documentation that it is clear why a special `BuildProfileRefID` is generated.
```
// BuildProfileRefID returns a UUID based off cluster, profileID, and product type to simplify SQL joins in SELECT statements.
```

Additionally we should forbid SQL updates on result fields to `clusterID`, `profileID` and product type that these fields are **not writable** anymore, instead if a user/dev wants to update these fields they need to delete all the respective results and let Central re-create them.

While this is clear to us now while implementing compliance v2, in longer-terms this code might be hard to understand because it solves a specific problem and introduces an implicit dependency on the `BuildProfileRefID` function.

TL;DR, I think we need:

- Slightly better documentation on this function to explain what it achieves and when someone should use it
- Prevent updates on `profileID`, `clusterID` and `product type`. (Whereas I see product type as critical when a name changes, e.g. RHACS to StackRox or OCP to OPP.) (Essentially just adding a bunch of if-statements to protect these fields)